plugins/inputs/prometheus/prometheus_test.go

```diff
@@ -577,3 +603,130 @@ func TestInitConfigSelectors(t *testing.T) {
 	require.NotNil(t, p.podLabelSelector)
 	require.NotNil(t, p.podFieldSelector)
 }
+
+func TestPrometheusInternalOk(t *testing.T) {
+	prommetric := `# HELP test_counter A sample test counter.
+# TYPE test_counter counter
+test_counter{label="test"} 1 1685443805885`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintln(w, prommetric)
+		require.NoError(t, err)
+	}))
+	defer ts.Close()
+
+	p := &Prometheus{
+		Log:                  testutil.Logger{},
+		KubernetesServices:   []string{ts.URL},
+		EnableRequestMetrics: true,
+	}
+	require.NoError(t, p.Init())
+
+	u, err := url.Parse(ts.URL)
+	require.NoError(t, err)
+	tsAddress := u.Hostname()
+
+	expected := []telegraf.Metric{
+		metric.New(
+			"prometheus_request",
+			map[string]string{
+				"address": tsAddress,
+				"result":  "success"},
+			map[string]interface{}{
+				"content_length":     int64(1),
+				"http_response_code": int64(200),
+				"response_time":      float64(0),
+				"result_code":        int64(0)},
+			time.UnixMilli(0),
+			telegraf.Untyped,
+		),
+	}
+
+	var acc testutil.Accumulator
+	fmt.Printf("%+v\n", acc.GetTelegrafMetrics())
+
+	require.NoError(t, acc.GatherError(p.Gather))
```

Comment:
Please get rid of the debug message
```suggestion
	var acc testutil.Accumulator
	require.NoError(t, acc.GatherError(p.Gather))
```